### ISSUE TITLE
[Bug]: BVT(race enabled) failed due to `no such table func_datetime_date.t1`

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#13853]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

2c5b04a97214b38330359af1f963c2d85a023eb1

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

<img width="1483" alt="image" src="https://github.com/matrixorigin/matrixone/assets/77312370/51994a1e-f1a4-4148-a7fd-e2f6b8359508">

https://github.com/matrixorigin/matrixone/actions/runs/7326932954/job/19953373667

### Steps to Reproduce

```Markdown
run bvt with race enabled
```

**Comment by sukki37** [comment]
repro: https://github.com/matrixorigin/matrixone/actions/runs/7329854338/job/19960405971

**Comment by nnsgmsone** [comment]
https://github.com/matrixorigin/matrixone/actions/runs/7326932954/job/19953373667
以上面这个为例，出问题的时候是报的错误信息是2023-12-26T07:52:44.2221171Z no such table func_datetime_timestampdiff.t1，导致这个的原因如下:
* 创建了一个表t1，然后往自增的表插入一行自增信息(t1的tableid)，可是在自增表里找不到自增这行的信息
* 可以确认表创建成功了，而且cn也看到了这个表，具体的日志如下：
		{"level":"INFO","time":"2023/12/26 07:52:44.103480 +0000","caller":"disttae/tools.go:832","msg":"precommit: create table: 32ef6b6fae164ba5a3820a5073b6f582/Active/S:1703577163992942201-1-[t1]"}
		{"level":"INFO","time":"2023/12/26 07:52:44.108483 +0000","caller":"rpc/handle.go:912","msg":"[precommit] create relation: {AccountID:0 UserID:1 RoleID:0}, 277253-func_datetime_timestampdiff:277259-t1 txn: ctx[32EF6B6FAE164BA5A3820A5073B6F582][1703577163992942201-1->9223372036854775807-4294967295][Active]: <nil>"}
		{"level":"INFO","time":"2023/12/26 07:52:44.170002 +0000","name":"log-service.incrservice","caller":"incrservice/service.go:268","msg":"try to get columns","uuid":"7c4dccb4-4d3c-41f8-b482-5251dc7a41bf","tableId":277259,"txn":"af053186103f4eb7a99f63abae8b993f/Active/S:1703577164165275524-1"}
	上述日志可以证明cn创建了这个表，dn也创建了这个表，然后cn也通过logtail看到了这个表，并且取到了对应的tableid 277259
	
* 可以确认自增表里的数据是缺失的，具体的日志如下:
		{"level":"ERROR","time":"2023/12/26 07:52:44.207369 +0000","caller":"incrservice/service.go:275","msg":"error: no such table .Cannot find tableID 277259 in table mo_increment_columns, accountid 0, txn: af053186103f4eb7a99f63abae8b993f/Active/S:1703577164165275524-1\tcol_name: __mo_fake_pk_col, table_id: 277258\n\tcol_name: __mo_fake_pk_col, table_id: 276986\n\tcol_name: task_id, table_id: 272389"}
	上述日志遍历了出问题时刻自增表里的数据，而且用的是同一个时间辍同一个快照(btree cow的快照)。

**Comment by LeftHandCold** [comment]
<img width="591" alt="image" src="https://github.com/matrixorigin/matrixone/assets/14086886/f334cdb6-cc20-4c43-be98-e9dd0c83406c">
<img width="909" alt="image" src="https://github.com/matrixorigin/matrixone/assets/14086886/37b63dcb-e8bc-45b6-850c-fcf8fb18390c">

**Comment by LeftHandCold** [comment]
{"level":"ERROR","time":"2023/12/26 21:53:39.300252 +0000","name":"log-service.incrservice","caller":"incrservice/service.go:96","msg":"create auto increment cache failed","uuid":"7c4dccb4-4d3c-41f8-b482-5251dc7a41bf","table-id":276954,"txn":"2470fc0b263545809bf45b9c67721800","error":"txn need retry in rc mode"}
两次出现no such table的场景都出现了 create auto increment cache重试的现象。
建表时在incrservice/store_sql.go中模拟一次retry的情况。可以必现

**Comment by LeftHandCold** [comment]
<img width="928" alt="image" src="https://github.com/matrixorigin/matrixone/assets/14086886/2264c9c9-5ec6-490c-8a90-7256f663e4b5">
<img width="788" alt="image" src="https://github.com/matrixorigin/matrixone/assets/14086886/401dfbdf-b61f-4b88-a05b-938c1c5919ea">

**Comment by zhangxu19830126** [comment]
fixed by #13912
